Sources/CollectionDiffing.swift

```diff
@@ -0,0 +1,560 @@
+import Foundation
+import Result
+
+/// Represents a snapshot of a collection.
+///
+/// The changeset associated with the first snapshot received by any given observation
+/// is generally irrelevant. Observations should special case the first snapshot as a
+/// complete replacement, or whatever semantic that fits their purpose.
+///
+/// `Snapshot` does not include a reference to the previous snapshot. If the
+/// history is relevant for delta consumption, it is up to the observers to cache the
+/// history on their own.
```

Comment:
Why not include that reference? With COW I wouldn't think it'd be a real performance concern.